src/runtime/js/ts.ui/core/core-api@tradeshift.com/models/dialogs/ts.ui.SpotlightModel.js

```diff
@@ -0,0 +1,188 @@
+				existingContent[0].remove();
+			}
+
+			if (existingButton.length) {
+				existingButton[0].remove();
+			}
+
+			var content = document.createElement('div');
+			content.style.position = 'absolute';
+			content.style.color = 'white';
+			content.className = 'ts-spotlight-content ts-transition';
+
+			content.style.left = targetPos.x + radius + 22 + 'px';
+			content.style.top = targetPos.y + (radius / 6) + 'px';
+
+			content.style.fontWeight = 'normal';
+			content.style.paddingRight = '10px';
+			content.style.zIndex = '3999';
+			content.style.width = '300px';
+
+			var closeButton = document.createElement('button');
+			closeButton.className = 'ts-spotlight-close-button ts-primary';
+			closeButton.setAttribute('data-ts', 'Button');
+			closeButton.style.position = 'absolute';
+
+			closeButton.style.top = '20px';
+			closeButton.style.right = '20px';
+			closeButton.style.width = '100px';
+			closeButton.style.height = '100px';
+			closeButton.style.opacity = 1;
+			closeButton.style.backgroundColor = 'transparent';
+			closeButton.style.zIndex = '999999';
+			closeButton.addEventListener('click', function(e) {
+				cover.fadeOut();
+				content.parentNode.removeChild(content);
+				closeButton.parentNode.removeChild(closeButton);
+				e.preventDefault();
+				return false;
+			}, true);
+
+			var iconButton = document.createElement('i');
+			iconButton.className = 'ts-icon-close';
+			iconButton.style.fontSize = '33px';
+
+			closeButton.appendChild(iconButton);
+
+			var title = document.createElement('h4');
+			title.className = 'ts-spotlight-title';
+			title.style.textTransform = 'uppercase';
+			title.innerHTML = titleText;
+
+			var text = document.createElement('p');
+			text.className = 'ts-spotlight-text ';
+			text.innerHTML = hintText;
+
+			content.appendChild(title);
+			content.appendChild(text);
+
+			document.getElementsByTagName('body')[0].appendChild(content);
+			document.getElementsByTagName('body')[0].appendChild(closeButton);
+
+			if (this.isOffScreen(content)) {
+				content.style.left = targetPos.x - (radius * 1.8) + 'px';
+				content.style.textAlign = 'right';
+			}
+
+			cover.position({
+				y: targetPos.y,
+				x: targetPos.x,
+				w: radius,
+				h: radius
+			});
+
+			cover.fadeIn();
+
+			content.className = 'ts-spotlight-content ts-transition ts-visible';
+
+			window.addEventListener('keydown', function(e) {
+				if ((e.key === 'Escape' || e.key === 'Esc' || e.keyCode === 27)) {
+					cover.fadeOut();
+					content.parentNode.removeChild(content);
+					closeButton.parentNode.removeChild(closeButton);
+					e.preventDefault();
+					return false;
+				}
+			}, true);
+
+			window.addEventListener('click', function(e) {
+				cover.fadeOut();
+				if (content.parentNode) {
+					content.parentNode.removeChild(content);
+				}
+				if (closeButton.parentNode) {
+					closeButton.parentNode.removeChild(closeButton);
+				}
+				e.preventDefault();
+				return false;
+			}, true);
+
```

Comment:
Similarly, there's a global broadcast called `gui.MOUSEEVENT_CLICK` or something that can be used here, so that the Spotlight also closes when the user is manhandling the menu menu (where the click event is in another `iframe` and thus ignored by the code above).